field/src/packed.rs

```diff
@@ -72,6 +72,7 @@ pub unsafe trait PackedField: AbstractField<F = Self::Scalar>
     /// 0 and it cannot exceed `WIDTH`.
     fn interleave(&self, other: Self, block_len: usize) -> (Self, Self);
 
+    // TODO: shouldn't this check alignment?
```

Comment:
?

Reply:
I can fix it, I just wanted to check if I was missing something

Reply:
I had also asked @nbgl this. My understanding is that it should be safe since the trait is implemented by arrays (or `repr(transparent)` wrappers around them), and arrays have the alignment of their element type -

> An array of `[T; N]` has a size of `size_of::<T>() * N` and the same alignment of `T`

(In this generic `PackedField` code, safe casting is just assumed, as per the safety notes on `PackedField`, so this is technically about implementations being safe/correct.)

In terms of performance, this small alignment might be bad in certain cases? Like if we load a `PackedBabyBearNeon` from memory into a vector register, it seems like the compiler might need to emit separate loads for each `BabyBear` element, to ensure the reads are aligned (on targets that require it). @nbgl mentioned she hasn't seen this issue come up in practice, but it might be something to watch out for.

Reply:
There's no arrays involved here, only slices, but it doesn't really matter: It's effectively casting from `&[u32]` to `&[u128]`, which can definitely be misaligned

Reply:
which is UB in C, at least

Reply:
ahh I see what you mean. “The alignment of array types is greater or equal to the alignment of its element type.” so it should *usually* be fine. maybe I’ll add a static assert if possible